solidity/test/rewards/TestRewardUpgrades.js

```diff
@@ -231,5 +230,32 @@ describe("Rewards/Upgrades", () => {
             // 50000 + 95000 = 145000
             expect(beneficiaryBalance).to.eq.BN(145000)
         })
+
+        it("correctly updates reward balances when withdrawing after finalizing upgrade", async () => {
+            await rewards.initiateRewardsUpgrade(
+                newRewards.address,
+                {from: owner}
+            )
+
+            await time.increase(2 * termLength + 1)  
+            
+            await rewards.setCloseTime(timestamps[2])
+            await rewards.finalizeRewardsUpgrade({from: owner})
+
+            await rewards.receiveReward(0, { from: beneficiary })
+            await rewards.receiveReward(1, { from: beneficiary })
+            await rewards.receiveReward(2, { from: beneficiary })
+
+            const totalRewards = await rewards.totalRewards()
+            const dispensedRewards = await rewards.dispensedRewards()
+            const unallocatedRewards = await rewards.unallocatedRewards()
+
+            // interval 0 allocates 50000
+            // interval 1 allocates 95000
+            // 50000 + 95000 = 145000
+            expect(totalRewards).to.eq.BN(145000)
```

Comment:
do you think we can add a comment here saying that `totalRewards = totalRewards - unallocatedRewards (before transfer)`?

Reply:
Hmm... I think it's better explained by the previous tests:
- _moves all unallocated rewards to new contract_ explains where `855000` tokens go
-  _correctly updates reward balances_ explains - looking at this and previous test - why `145000` tokens stay in the contract

And the assertion pointed in this comment is inherited from that test. What we mostly care is `dispensedRewards` new value.